File: iOS/JobLoyal/UIKit/App/Main/Message/MessageTableViewController/Model/MessageModel.swift
//
//  MessageModel.swift
//  JobLoyal
//
//  Created by Sina khanjani on 3/29/1400 AP.
//

import Foundation

// MARK: - DataClass
struct MessagesModel: Codable, Hashable {
    let items: [MessageModel]
}

// MARK: - Item
struct MessageModel: Codable, Hashable {
    let id, userID, subject, itemDescription: String?
    let createdAt, updatedAt: String?
    let reply: MessageReplyModel?

    enum CodingKeys: String, CodingKey {
        case id
        case userID = "user_id"
        case subject
        case itemDescription = "description"
        case createdAt, updatedAt, reply
    }
}

// MARK: - RCSendCommentModel
struct MessageReplyModel: Codable, Hashable {
    let id, messageID, answer, createdAt: String?
    let updatedAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case messageID = "message_id"
        case answer, createdAt, updatedAt
    }
}

File: android/app/src/main/java/com/jobloyal/jobber/model/message/MessageModel.kt
package com.jobloyal.jobber.model.message

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class MessageModel(
    val createdAt: String?,
    val description: String?,
    val id: String?,
    val reply: Reply?,
    val subject: String?,
    val updatedAt: String?,
    val user_id: String?
) : Parcelable